osu.Game/Rulesets/Scoring/HealthProcessor.cs

```diff
@@ -19,7 +22,7 @@ public abstract partial class HealthProcessor : JudgementProcessor
         /// <summary>
         /// Additional conditions on top of <see cref="CheckDefaultFailCondition"/> that cause a failing state.
         /// </summary>
-        public event Func<HealthProcessor, JudgementResult, bool>? FailConditions;
+        public List<IHasFailCondition> FailConditions = new List<IHasFailCondition>();
```

Comment:
At this point I suggest accepting a list of all selected mods from `Player` similar to how `ScoreProcessor` does it, and then iterate through them in `meetsAnyFailCondition` looking for any mod that implements `IHasFailCondition` and check accordingly.